presets/ragengine/vector_store/base.py

```diff
@@ -381,20 +422,21 @@ async def chat_completion(self, request: dict) -> ChatCompletionResponse:
         logger.info(
             f"Creating chat engine for index '{request.get('index_name')}' with prompt size: {prompt_len}"
         )
-        chat_engine = self.index_map[
-            request.get("index_name")
-        ].as_chat_engine(
+        # Calculate top_k based on available context. For larger windows, we can afford to retrieve more documents.
+        # 750 tokens is a rough estimate based off max 1500 for code and 1000 for text
+        top_k = max(100, int((self.llm.metadata.context_window - prompt_len) / 750))
+        chat_engine = self.index_map[request.get("index_name")].as_chat_engine(
             llm=self.llm,
-            similarity_top_k=100,  # Might want to make this a function of avg doc node size in an index but this should be a wide enough default
-            chat_mode=ChatMode.CONDENSE_PLUS_CONTEXT,
+            similarity_top_k=top_k,
+            chat_mode=ChatMode.CONTEXT,
             node_postprocessors=[
                 ContextSelectionProcessor(
                     rag_context_token_fill_ratio=request.get(
                         "context_token_ratio", RAG_DEFAULT_CONTEXT_TOKEN_FILL_RATIO
                     ),
                     llm=self.llm,
                     max_tokens=max_tokens,
-                    similarity_threshold=0.8,
+                    similarity_threshold=0.85,
```

Comment:
Why change this? I'd prefer having a comment for every hardcoded number in the code base.

Reply:
The similarity search values resulting from FAISS search are not consistent but are usually within some range. The [test here](https://github.com/kaito-project/kaito/blob/fe7648fd4cacd3bbef9c6ceefe90734383dc10ae/presets/ragengine/tests/api/test_node_processing.py#L44) was flaky with some similarity score hovering > 0.8 causing it to fail. I'll make an env var for this one as well, I think it might be work having the ability to tweak

Reply:
I already had this as a env var at 0.85 but forgot to swap it. Moved to using the env var as well as created another env var for document node token approximation.